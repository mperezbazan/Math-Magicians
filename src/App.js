function App() {
  return (
    <div className="App">
      Temp
    </div>
  );
}

export default App;
