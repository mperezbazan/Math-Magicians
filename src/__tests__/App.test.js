import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import App from '../App';

test('Render App Container ', async () => {
  render(<Router><App /></Router>);
  await userEvent.click(screen.getByText('Home'));
  expect(screen.getByRole('heading')).toHaveTextContent('Welcome to our page!');
});

test('Match App Snapshot', () => {
  const tree = renderer.create(<Router><App /></Router>).toJSON();
  expect(tree).toMatchSnapshot();
});
