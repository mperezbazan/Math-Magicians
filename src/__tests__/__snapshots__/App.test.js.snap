// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match App Snapshot 1`] = `
Array [
  <header
    className="header-container"
  >
    <div
      className="logo-container"
    >
      <p
        className="logo-text"
      >
        Math Magicians
      </p>
    </div>
    <nav
      className="nav-container"
    >
      <ul
        className="nav-links-container"
      >
        <li
          className="nav-link-item"
        >
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="nav-link-item"
        >
          <a
            className=""
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li
          className="nav-link-item"
        >
          <a
            className=""
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <div
    className="home-container"
  >
    <h1
      className="page-title"
    >
      Welcome to our page!
    </h1>
    <p
      className="home-text"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent et nisi vitae augue sagittis rutrum nec at est. In augue mi, venenatis eget mi ultrices, varius iaculis enim. Ut posuere, leo in congue ullamcorper, nulla lacus commodo ligula, et faucibus mauris sem quis sapien. Etiam hendrerit justo mauris, quis egestas mi maximus vel. Aliquam in purus convallis, commodo est at, ullamcorper elit. Proin consectetur, dolor in tristique placerat, est quam posuere urna, dapibus placerat nulla ex eu urna. Donec eu fringilla odio. Donec consequat ut dui at mattis. Sed bibendum lobortis orci et efficitur. Maecenas in neque molestie nisl pellentesque sodales. Donec fringilla eget magna et cursus. Suspendisse eu lorem est. Vestibulum scelerisque varius vulputate. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.l
    </p>
    <p
      className="home-text"
    >
      Sed non ipsum sapien. Pellentesque blandit sodales justo pretium tristique. Curabitur sagittis diam velit, at auctor augue euismod ut. Duis efficitur eu neque sit amet iaculis. Nulla sagittis auctor imperdiet. Vestibulum sit amet lorem imperdiet, consequat nisl in, vehicula ante. Morbi eget luctus est. Aenean facilisis nunc vel sem volutpat maximus. Morbi vel maximus nunc. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Mauris ut ullamcorper nisi, consectetur mattis urna. Quisque arcu leo, condimentum non tincidunt et, sollicitudin et augue.
    </p>
  </div>,
]
`;
